daemon/config/builder_test.go

```diff
@@ -42,3 +42,24 @@ func TestBuilderGC(t *testing.T) {
 	assert.Assert(t, filters.Args(cfg.Builder.GC.Policy[0].Filter).UniqueExactMatch("unused-for", "2200h"))
 	assert.Assert(t, filters.Args(cfg.Builder.GC.Policy[1].Filter).UniqueExactMatch("unused-for", "3300h"))
 }
+
+// Regression test for https://github.com/moby/moby/issues/44361
+func TestBuilderGCPanic(t *testing.T) {
```

Comment:
Already LGTM'd, but I think it would make sense to name this test based on what is being tested which clearly was not being tested before.
We aren't testing to see if the builder panics, we are testing how it handles various inputs.